src/idl_gen_cpp.cpp

```diff
@@ -708,10 +709,20 @@ struct IsAlnum {
   }
 };
 
-// Iterate through all definitions we haven't generate code for (enums, structs,
-// and tables) and output them to a single file.
-std::string GenerateCPP(const Parser &parser,
-                        const std::string &file_name) {
+static std::string GeneratedFileName(const std::string &path,
+                                     const std::string &file_name) {
+  return path + file_name + "_generated.h";
+}
+
+namespace cpp {
+class CppGenerator : public BaseGenerator {
+public:
+  CppGenerator(const Parser &parser_, const std::string &path_,
+               const std::string &file_name_)
+      : BaseGenerator(parser_, path_, file_name_){};
+  // Iterate through all definitions we haven't generate code for (enums, structs,
+  // and tables) and output them to a single file.
+  bool generate() {
   // Check if we have any code to generate at all, to avoid an empty header.
   for (auto it = parser.enums_.vec.begin(); it != parser.enums_.vec.end();
```

Comment:
I cut this code and pasted it there (this is moved old code). 
As I didn't want clang-format to mess this old code, so that git could correctly display was has changed and what hasn't (it still doesn't look like I thought it would look though).

I'll indent it with clang-format then (everything should be indented 2 spaces to the right as this function was moved inside a class) but I fear that it will mess things for the pr. :/